tuxemon/states/character/__init__.py

```diff
@@ -49,175 +59,161 @@ def add_menu_items(
             results = db.lookup(mon, table="monster")
             if results.txmn_id > 0:
                 filters.append(results)
-        tuxepedia = list(player.tuxepedia.values())
+        tuxepedia = list(self.char.tuxepedia.values())
         caught = tuxepedia.count(SeenStatus.caught)
         seen = tuxepedia.count(SeenStatus.seen) + caught
         percentage = round((seen / len(filters)) * 100, 1)
 
-        msg_progress = T.format(
-            "tuxepedia_progress", {"value": str(percentage)}
-        )
-        msg_seen = T.format(
-            "tuxepedia_data_seen",
-            {"param": str(seen), "all": str(len(filters))},
-        )
-        msg_caught = T.format(
-            "tuxepedia_data_caught",
-            {"param": str(caught), "all": str(len(filters))},
-        )
-        date_begin = formula.today_ordinal() - int(
-            player.game_variables["date_start_game"]
-        )
+        _msg_progress = {"value": str(percentage)}
+        msg_progress = T.format("tuxepedia_progress", _msg_progress)
+        _msg_seen = {"param": str(seen), "all": str(len(filters))}
+        msg_seen = T.format("tuxepedia_data_seen", _msg_seen)
+        _msg_caught = {"param": str(caught), "all": str(len(filters))}
+        msg_caught = T.format("tuxepedia_data_caught", _msg_caught)
+
+        today = formula.today_ordinal()
+        date = self.char.game_variables.get("date_start_game", today)
+        date_begin = today - int(date)
         msg_begin = (
             T.format("player_start_adventure", {"date": date_begin})
             if date_begin >= 1
             else T.translate("player_start_adventure_today")
         )
-        tot = len(player.battles)
-        won = sum(
-            1
-            for battle in player.battles
-            if battle.outcome == OutputBattle.won
-        )
-        draw = sum(
-            1
-            for battle in player.battles
-            if battle.outcome == OutputBattle.draw
-        )
-        lost = sum(
-            1
-            for battle in player.battles
-            if battle.outcome == OutputBattle.lost
-        )
-        msg_battles = T.format(
-            "player_battles",
-            {
-                "tot": str(tot),
-                "won": str(won),
-                "draw": str(draw),
-                "lost": str(lost),
-            },
-        )
+
+        battles = self.char.battles
+        tot = len(battles)
+        _won = OutputBattle.won
+        _draw = OutputBattle.draw
+        _lost = OutputBattle.lost
+        won = sum(1 for battle in battles if battle.outcome == _won)
```

Comment:
this is fine for now, but if battles becomes very large, then we would want to break this down into something like:
```python
for battle in battles
  if battle.outcome == OutputBattle.won:
     won += 1
  ...
```
because it would loop over the battle list only once.  probably not realistic here, but worth think about it more because iteration of things in python can be slow.
also, this could this be rewritten as `len(battle for battle in battles if battle.outcome == _won)`?  i don't think we need sum here.

Reply:
totally agree with you, implemented